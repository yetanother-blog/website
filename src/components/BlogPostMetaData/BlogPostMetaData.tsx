import React from 'react';
import { Typography } from '../Typography/Typography';
import { useTheme, DefaultTheme } from 'styled-components';

interface BlogPostMetaDataProps {
  date: React.ReactNode;
  timeToRead: React.ReactNode;
  author: React.ReactNode;
  mb?: keyof DefaultTheme['space'];
}

export const BlogPostMetaData: React.FC<BlogPostMetaDataProps> = ({
  date,
  timeToRead,
  author,
  mb = 'l'
}) => {
  const theme = useTheme();
  return (
    <Typography
      variant="tinyText"
      fontFamily="Source Code Pro"
      fontWeight="600"
      mb={theme.space[mb]}
    >
      {date} • {author} • {timeToRead}min read
    </Typography>
  );
};
