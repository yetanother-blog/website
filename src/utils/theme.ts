import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
