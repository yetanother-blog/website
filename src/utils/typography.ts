import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
