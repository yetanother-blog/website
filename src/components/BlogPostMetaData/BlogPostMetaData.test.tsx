import React from 'react';
import { render } from '@testing-library/react';
import { BlogPostMetaData } from './BlogPostMetaData';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('BlogPostMetaData component', () => {
  it('should render a BlogPostMetaData w/ content', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <BlogPostMetaData date="May 25, 2020" wordCount="1337" timeToRead="4" />
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
