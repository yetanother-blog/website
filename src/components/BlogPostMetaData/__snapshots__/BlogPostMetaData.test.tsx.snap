// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostMetaData component should render a BlogPostMetaData w/ content 1`] = `
<div>
  <span
    class="sc-jRQAMF dSLHev"
    display="block"
    font-family="Source Code Pro"
    font-weight="600"
  >
    <time
      datetime="2020-05-25 15:00:00.000Z"
    >
      May 25, 2020
    </time>
     •
     
    <address
      class="sc-gKckTs bsCFOL"
      display="inline-block"
    >
      Homer Simpson
    </address>
     
    •
     
    <span
      class="sc-gKckTs bsCFOL"
      display="inline-block"
    >
      4min read
    </span>
  </span>
</div>
`;
